### DISCUSSION TITLE
Does the row header support sorting and filtering functions?

### DISCUSSION THREAD

**Discussion by sanshigongA** [eclipse-birt/birt#1759] [Q&A]

**Comment by speckyspooky** [comment]
The BIRT-preview is a classic and static HTML-page without any grouping, sortig or filter option.
The main target is to simulate the print preview of the document and therefore there are no further dashboard functions added.

So you have to handle the situation that normally a page break would be used on huge data.
You have to avoid that and in some cases it could be not enough because the maximum of very huge data will arrive the maximum page limit and you would get a page break again (ok, the second case needs relly large data).

If you want it than you have 3 options:
- add the JavaScripts with classic implementation of JS-functions
- add specialized JS libraries for the HTML page and add so the functionality
- contribute the functionality for the BIRD project

**Reply by SteveSchafer-Innovent** [reply]
If BIRT-preview is used such that the entire report is a single HTML document then it would just be a matter of using browser-side javascript to manipulate the DOM tree to sort and filter the table.  You'd also need to implement a user interface.  It wouldn't be trivial but quite doable.

If using the web-viewer you have to contend with pages.  You could try to force the entire table to be on one page and then use the same technique as above.  Or a more complex but perhaps better solution would be to tell the emitter to re-generate the page and possibly subsequent pages.  I don't even know if that would be possible but it would be fun to find out.

Doing this for excel and pdf output is probably impossible.

**Reply by speckyspooky** [reply]
One of the library options would be JS-DataTables [https://datatables.net/](url)
But also there some special options like filtering are DataTable-PlugIns.

I work at my BIRT-platform with this library to create DashBoards with dynamic tables.
